components/suggestion/search_channel_with_permissions_provider.jsx

```diff
@@ -0,0 +1,237 @@
+
+        const channelsAsync = ChannelActions.autocompleteChannelsForSearch(teamId, channelPrefix)(this.dispatch, this.getState);
+
+        let channelsFromServer = [];
+        try {
+            const {data} = await channelsAsync;
+            channelsFromServer = data;
+        } catch (err) {
+            AppDispatcher.handleServerAction({
+                type: ActionTypes.RECEIVED_ERROR,
+                err,
+            });
+        }
+
+        if (this.shouldCancelDispatch(channelPrefix)) {
+            return;
+        }
+
+        const channels = getChannelsInCurrentTeam(state).concat(channelsFromServer);
+        this.formatChannelsAndDispatch(channelPrefix, resultsCallback, channels);
+    }
+
+    formatChannelsAndDispatch(channelPrefix, resultsCallback, allChannels) {
+        const channels = [];
+
+        const state = this.getState();
+
+        const members = getMyChannelMemberships(state);
+
+        if (this.shouldCancelDispatch(channelPrefix)) {
+            return;
+        }
+
+        const completedChannels = {};
+
+        const channelFilter = this.makeChannelSearchFilter(channelPrefix);
+
+        const config = getConfig(state);
+        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
+
+        for (const id of Object.keys(allChannels)) {
+            const channel = allChannels[id];
+
+            if (completedChannels[channel.id]) {
+                continue;
+            }
+
+            if (channelFilter(channel)) {
+                const newChannel = Object.assign({}, channel);
+                const channelIsArchived = channel.delete_at !== 0;
+
+                const wrappedChannel = {channel: newChannel, name: newChannel.name, deactivated: false};
+                if (!viewArchivedChannels && channelIsArchived) {
+                    continue;
+                } else if (!members[channel.id]) {
+                    continue;
+                } else if (channelIsArchived && !members[channel.id]) {
+                    continue;
+                } else if (channel.type === Constants.OPEN_CHANNEL) {
+                    wrappedChannel.type = Constants.OPEN_CHANNEL;
+                } else if (channel.type === Constants.PRIVATE_CHANNEL) {
+                    wrappedChannel.type = Constants.PRIVATE_CHANNEL;
+                } else {
+                    continue;
+                }
+                completedChannels[channel.id] = true;
+                channels.push(wrappedChannel);
+            }
+        }
+
+        const channelNames = channels.
+            sort(channelSearchSorter).
+            map((wrappedChannel) => wrappedChannel.channel.name);
+
+        resultsCallback({
+            matchedPretext: channelPrefix,
+            terms: channelNames,
+            items: channels,
+            component: SearchChannelWithPermissionsSuggestion,
+        });
+    }
+
+    getState = store.getState;
+
+    dispatch = store.dispatch;
+
+    appDispatch = (results) => {
```

Comment:
I don't see where `appDispatch` is used.  Is this necessary?

Reply:
I missed this when updating. I'm removing these three properties as I modify the test to follow.